### PR TITLE
Extract JanusGraph Gremlin driver requirements

### PR DESCRIPTION
**Pull request by farodin91**
Support things
* Predicates
* Geoshape
* RelationIdenitifier

ToDo:
- [x] GraphSON against a janusgraph instance
  * [x] geoshape
  * [x] vertices
  * [x] relationIdentifier
  * [x] predicates
- [x] Gyro against a janusgraph instance
  * [x] geoshape
  * [x] vertices
  * [x] relationIdentifier
  * [x] predicates

Signed-off-by: Jan Jansen <jan.jansen@gdata.de>

Discussion on google groups: https://groups.google.com/forum/#!topic/janusgraph-dev/nmdEk_uGxdI

-----

Thank you for contributing to JanusGraph!

In order to streamline the review of the contribution we ask you
to ensure the following steps have been taken:

### For all changes:
- [x] Is there an issue associated with this PR? Is it referenced in the commit message?
- [x] Does your PR body contain #xyz where xyz is the issue number you are trying to resolve?
- [x] Has your PR been rebased against the latest commit within the target branch (typically `master`)?
- [x] Is your initial contribution a single, squashed commit?

### For code changes:
- [x] Have you written and/or updated unit tests to verify your changes?
- [x] If adding new dependencies to the code, are these dependencies licensed in a way that is compatible for inclusion under [ASF 2.0](http://www.apache.org/legal/resolved.html#category-a)?
- [x] If applicable, have you updated the LICENSE.txt file, including the main LICENSE.txt file in the root of this repository?
- [x] If applicable, have you updated the NOTICE.txt file, including the main NOTICE.txt file found in the root of this repository?

### For documentation related changes:
- [x] Have you ensured that format looks appropriate for the output in which it is rendered?
- [x] If this PR is a documentation-only change, have you added a `[skip ci]`
  tag to the first line of your commit message to avoid spending CPU cycles in
  Travis CI when no code, tests, or build configuration are modified?

### Note:
Please ensure that once the PR is submitted, you check Travis CI for build issues and submit an update to your PR as soon as possible.

Any thoughts?

### CODE REVIEW

**Comment by porunov** [comment]
@farodin91 I will try to review it this week

**Comment by farodin91** [comment]
@porunov @FlorianHockmann  Do you like to review this PR?

**Comment by porunov** [comment]
@farodin91 I didn't review this PR yet but I see that Travis CI fails and Codacy has a lot of issues. I think that should be fixed.

**Comment by farodin91** [comment]
> @farodin91 I didn't review this PR yet but I see that Travis CI fails and Codacy has a lot of issues. I think that should be fixed.

I don't want to fix all codacy issue to prevent any major changes.

**Comment by porunov** [comment]
> I don't want to fix all codacy issue to prevent any major changes

That is OK but I think at least CI build should pass

**Comment by farodin91** [comment]
@porunov I did break something by rebasing.

File: janusgraph-core/src/main/java/org/janusgraph/core/attribute/GeoshapeSerializer.java [hunk 1]
```diff
@@ -0,0 +1,203 @@
...
+    @Override
+    public void write(WriteBuffer buffer, Geoshape attribute) {
+        try {
+            ByteArrayOutputStream outputStream = new ByteArrayOutputStream();
+            Geoshape.GeoshapeBinarySerializer.write(outputStream, attribute);
+            byte[] bytes = outputStream.toByteArray();
+            VariableLong.writePositive(buffer,bytes.length);
+            buffer.putBytes(bytes);
+        } catch (IOException e) {
+            throw new RuntimeException("I/O exception writing geoshape", e);
```
Comment (FlorianHockmann) [inline janusgraph-core/src/main/java/org/janusgraph/core/attribute/GeoshapeSerializer.java]: Codacy Issue found: [Avoid throwing raw exception types.](https://app.codacy.com/app/JanusGraph/janusgraph/pullRequest?prid=3377550)

**Comment by FlorianHockmann** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
+ Solved 2
- Added 8
           

Complexity increasing per file
==============================
- janusgraph-core/src/main/java/org/janusgraph/core/attribute/GeoshapeSerializer.java  18
- janusgraph-driver/src/test/java/org/janusgraph/graphdb/tinkerpop/io/graphson/JanusGraphSONModuleIT.java  1
- janusgraph-core/src/main/java/org/janusgraph/graphdb/relations/RelationIdentifierUtils.java  10
- janusgraph-driver/src/test/java/org/janusgraph/core/attribute/TinkerPopTextWrappingPredicateTest.java  1
- janusgraph-driver/src/main/java/org/janusgraph/graphdb/tinkerpop/plugin/JanusGraphGremlinDriverPlugin.java  3
- janusgraph-driver/src/main/java/org/janusgraph/graphdb/relations/RelationIdentifier.java  9
- janusgraph-driver/src/test/java/org/janusgraph/graphdb/tinkerpop/JanusGraphIoRegistryTest.java  2
- janusgraph-driver/src/test/java/org/janusgraph/graphdb/tinkerpop/JanusGraphGryoIT.java  1
- janusgraph-driver/src/test/java/org/janusgraph/graphdb/tinkerpop/JanusGraphSerializerBaseIT.java  1
- janusgraph-driver/src/test/java/org/janusgraph/graphdb/tinkerpop/plugin/JanusGraphGremlinDriverPluginTest.java  3
- janusgraph-driver/src/test/java/org/janusgraph/util/encoding/LongEncodingTest.java  3
- janusgraph-driver/src/main/java/org/janusgraph/core/attribute/Cmp.java  2
- janusgraph-core/src/main/java/org/janusgraph/graphdb/query/JanusGraphPredicateUtils.java  14
- janusgraph-driver/src/test/java/org/janusgraph/graphdb/tinkerpop/io/graphson/JanusGraphSONModuleTest.java  2
- janusgraph-driver/src/test/java/org/janusgraph/core/JanusGraphContainer.java  1
- janusgraph-driver/src/main/java/org/janusgraph/core/attribute/Geoshape.java  7
- janusgraph-driver/src/main/java/org/janusgraph/graphdb/tinkerpop/JanusGraphIoRegistryV1d0.java  1
- janusgraph-driver/src/main/java/org/janusgraph/graphdb/database/serialize/AttributeUtils.java  12
- janusgraph-driver/src/test/java/org/janusgraph/core/attribute/GeoshapeHelperTest.java  1
- janusgraph-core/src/main/java/org/janusgraph/graphdb/database/serialize/InternalAttributeUtil.java  1
- janusgraph-driver/src/main/java/org/janusgraph/graphdb/tinkerpop/JanusGraphIoRegistry.java  1
         

Complexity decreasing per file
==============================
+ janusgraph-core/src/main/java/org/janusgraph/graphdb/tinkerpop/plugin/JanusGraphGremlinPlugin.java  -2
         

Clones added
============
- janusgraph-driver/src/main/java/org/janusgraph/graphdb/relations/RelationIdentifier.java  2
- janusgraph-driver/src/test/java/org/janusgraph/graphdb/tinkerpop/JanusGraphIoRegistryTest.java  1
- janusgraph-driver/src/test/java/org/janusgraph/graphdb/tinkerpop/plugin/JanusGraphGremlinDriverPluginTest.java  1
- janusgraph-driver/src/test/java/org/janusgraph/graphdb/tinkerpop/io/graphson/JanusGraphSONModuleTest.java  1
- janusgraph-driver/src/test/java/org/janusgraph/core/attribute/GeoshapeHelperTest.java  2
         
```


See the complete overview on [Codacy](https://app.codacy.com/app/JanusGraph/janusgraph/pullRequest?prid=3377550&bid=12157851)

File: janusgraph-driver/src/main/java/org/janusgraph/graphdb/tinkerpop/plugin/JanusGraphGremlinDriverPlugin.java [hunk 2]
```diff
@@ -0,0 +1,157 @@
...
+            })
+            .filter(JanusGraphGremlinDriverPlugin::isMethodStatic)
+            .forEach(METHOD_IMPORTS::add);
+
+        Stream.of(Text.values())
+            .map(v -> {
+                try {
+                    return Text.class.getMethod(v.toString(), Object.class);
+                } catch (NoSuchMethodException e) {
+                    throw new RuntimeException(e);
```
Comment (FlorianHockmann) [inline janusgraph-driver/src/main/java/org/janusgraph/graphdb/tinkerpop/plugin/JanusGraphGremlinDriverPlugin.java]: Codacy Issue found: [Avoid throwing raw exception types.](https://app.codacy.com/app/JanusGraph/janusgraph/pullRequest?prid=3377550)

File: janusgraph-core/src/main/java/org/janusgraph/graphdb/relations/RelationIdentifierUtils.java [hunk 3]
```diff
@@ -0,0 +1,86 @@
...
+
+    private static Iterable<? extends JanusGraphRelation> getJanusGraphRelations(RelationIdentifier rId, JanusGraphTransaction tx, JanusGraphVertex v, RelationType typeVertex) {
+        if (typeVertex.isEdgeLabel()) {
+            return findEdgeRelations(v, typeVertex, rId, tx);
+        } else {
+            return ((VertexCentricQueryBuilder) v.query()).noPartitionRestriction().types((PropertyKey) typeVertex).properties();
+        }
+    }
+
+    public static Iterable<? extends JanusGraphRelation> findEdgeRelations(JanusGraphVertex v, RelationType type, RelationIdentifier rId, JanusGraphTransaction tx){
```
Comment (FlorianHockmann) [inline janusgraph-core/src/main/java/org/janusgraph/graphdb/relations/RelationIdentifierUtils.java]: Codacy Issue found: [Avoid reassigning parameters such as 'v'](https://app.codacy.com/app/JanusGraph/janusgraph/pullRequest?prid=3377550)

File: janusgraph-core/src/main/java/org/janusgraph/graphdb/relations/RelationIdentifierUtils.java [hunk 4]
```diff
@@ -0,0 +1,86 @@
...
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package org.janusgraph.graphdb.relations;
+
+import org.apache.tinkerpop.gremlin.structure.Direction;
+import org.janusgraph.core.*;
```
Comment (FlorianHockmann) [inline janusgraph-core/src/main/java/org/janusgraph/graphdb/relations/RelationIdentifierUtils.java]: Codacy Issue found: [Avoid unused imports such as 'org.janusgraph.core'](https://app.codacy.com/app/JanusGraph/janusgraph/pullRequest?prid=3377550)

File: janusgraph-core/src/main/java/org/janusgraph/graphdb/transaction/StandardJanusGraphTx.java [hunk 5]
```diff
@@ -31,10 +31,7 @@
 import org.janusgraph.diskstorage.EntryList;
 import org.janusgraph.diskstorage.keycolumnvalue.SliceQuery;
 import org.janusgraph.graphdb.query.profile.QueryProfiler;
-import org.janusgraph.graphdb.relations.RelationComparator;
-import org.janusgraph.graphdb.relations.RelationIdentifier;
-import org.janusgraph.graphdb.relations.StandardEdge;
-import org.janusgraph.graphdb.relations.StandardVertexProperty;
+import org.janusgraph.graphdb.relations.*;
```
Comment (FlorianHockmann) [inline janusgraph-core/src/main/java/org/janusgraph/graphdb/transaction/StandardJanusGraphTx.java]: Codacy Issue found: [Avoid unused imports such as 'org.janusgraph.graphdb.relations'](https://app.codacy.com/app/JanusGraph/janusgraph/pullRequest?prid=3377550)

File: janusgraph-core/src/main/java/org/janusgraph/core/attribute/GeoshapeSerializer.java [hunk 6]
```diff
@@ -0,0 +1,203 @@
...
+ */
+public class GeoshapeSerializer implements AttributeSerializer<Geoshape> {
+
+    @Override
+    public void verifyAttribute(Geoshape value) {
+        //All values of Geoshape are valid
+    }
+
+    @Override
+    public Geoshape convert(Object value) {
```
Comment (FlorianHockmann) [inline janusgraph-core/src/main/java/org/janusgraph/core/attribute/GeoshapeSerializer.java]: Codacy Issue found: [Avoid reassigning parameters such as 'value'](https://app.codacy.com/app/JanusGraph/janusgraph/pullRequest?prid=3377550)

File: janusgraph-core/src/main/java/org/janusgraph/core/attribute/GeoshapeSerializer.java [hunk 7]
```diff
@@ -0,0 +1,203 @@
...
+        } catch (IOException e) {
+            // retry using legacy point deserialization
+            try {
+                ((ReadArrayBuffer) buffer).movePositionTo(position);
+                final float lat = buffer.getFloat();
+                final float lon = buffer.getFloat();
+                return Geoshape.point(lat, lon);
+            } catch (Exception ignored) { }
+            // throw original exception
+            throw new RuntimeException("I/O exception reading geoshape", e);
```
Comment (FlorianHockmann) [inline janusgraph-core/src/main/java/org/janusgraph/core/attribute/GeoshapeSerializer.java]: Codacy Issue found: [Avoid throwing raw exception types.](https://app.codacy.com/app/JanusGraph/janusgraph/pullRequest?prid=3377550)

File: janusgraph-core/src/main/java/org/janusgraph/core/attribute/GeoshapeSerializer.java [hunk 8]
```diff
@@ -0,0 +1,203 @@
...
+        } catch (ClassCastException | IOException | ParseException e) {
+            throw new IllegalArgumentException("GeoJSON was unparsable");
+        }
+    }
+
+    private Geoshape convertGeometry(Map<String, Object> geometry) throws IOException, ParseException {
+        String type = (String) geometry.get("type");
+        List<Object> coordinates = (List) geometry.get(Geoshape.FIELD_COORDINATES);
+
+        switch (type) {
```
Comment (FlorianHockmann) [inline janusgraph-core/src/main/java/org/janusgraph/core/attribute/GeoshapeSerializer.java]: Codacy Issue found: [Switch statements should have a default label](https://app.codacy.com/app/JanusGraph/janusgraph/pullRequest?prid=3377550)

**Comment by FlorianHockmann** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
+ Solved 2
- Added 8
           

Complexity increasing per file
==============================
- janusgraph-driver/src/main/java/org/janusgraph/graphdb/database/serialize/AttributeUtils.java  12
- janusgraph-core/src/main/java/org/janusgraph/graphdb/relations/RelationIdentifierUtils.java  10
- janusgraph-driver/src/test/java/org/janusgraph/graphdb/tinkerpop/JanusGraphIoRegistryTest.java  2
- janusgraph-driver/src/test/java/org/janusgraph/graphdb/tinkerpop/JanusGraphSerializerBaseIT.java  1
- janusgraph-driver/src/test/java/org/janusgraph/core/attribute/TinkerPopTextWrappingPredicateTest.java  1
- janusgraph-driver/src/test/java/org/janusgraph/util/encoding/LongEncodingTest.java  3
- janusgraph-driver/src/main/java/org/janusgraph/graphdb/relations/RelationIdentifier.java  9
- janusgraph-driver/src/test/java/org/janusgraph/graphdb/tinkerpop/plugin/JanusGraphGremlinDriverPluginTest.java  3
- janusgraph-driver/src/test/java/org/janusgraph/graphdb/tinkerpop/io/graphson/JanusGraphSONModuleTest.java  2
- janusgraph-core/src/main/java/org/janusgraph/graphdb/query/JanusGraphPredicateUtils.java  14
- janusgraph-driver/src/main/java/org/janusgraph/core/attribute/Cmp.java  2
- janusgraph-driver/src/main/java/org/janusgraph/graphdb/tinkerpop/JanusGraphIoRegistryV1d0.java  1
- janusgraph-driver/src/test/java/org/janusgraph/graphdb/tinkerpop/io/graphson/JanusGraphSONModuleIT.java  1
- janusgraph-driver/src/main/java/org/janusgraph/graphdb/tinkerpop/JanusGraphIoRegistry.java  1
- janusgraph-driver/src/test/java/org/janusgraph/core/attribute/GeoshapeHelperTest.java  1
- janusgraph-core/src/main/java/org/janusgraph/graphdb/database/serialize/InternalAttributeUtil.java  1
- janusgraph-driver/src/test/java/org/janusgraph/graphdb/tinkerpop/JanusGraphGryoIT.java  1
- janusgraph-driver/src/main/java/org/janusgraph/graphdb/tinkerpop/plugin/JanusGraphGremlinDriverPlugin.java  3
- janusgraph-driver/src/test/java/org/janusgraph/core/JanusGraphContainer.java  1
- janusgraph-core/src/main/java/org/janusgraph/core/attribute/GeoshapeSerializer.java  18
- janusgraph-driver/src/main/java/org/janusgraph/core/attribute/Geoshape.java  7
         

Complexity decreasing per file
==============================
+ janusgraph-core/src/main/java/org/janusgraph/graphdb/tinkerpop/plugin/JanusGraphGremlinPlugin.java  -2
         

Clones added
============
- janusgraph-driver/src/test/java/org/janusgraph/graphdb/tinkerpop/JanusGraphIoRegistryTest.java  1
- janusgraph-driver/src/main/java/org/janusgraph/graphdb/relations/RelationIdentifier.java  2
- janusgraph-driver/src/test/java/org/janusgraph/graphdb/tinkerpop/plugin/JanusGraphGremlinDriverPluginTest.java  1
- janusgraph-driver/src/test/java/org/janusgraph/graphdb/tinkerpop/io/graphson/JanusGraphSONModuleTest.java  1
- janusgraph-driver/src/test/java/org/janusgraph/core/attribute/GeoshapeHelperTest.java  2
         
```


See the complete overview on [Codacy](https://app.codacy.com/app/JanusGraph/janusgraph/pullRequest?prid=3377550&bid=12157851)

File: janusgraph-driver/src/main/java/org/janusgraph/graphdb/tinkerpop/plugin/JanusGraphGremlinDriverPlugin.java (hunk 2 above)
Comment (FlorianHockmann) [inline janusgraph-driver/src/main/java/org/janusgraph/graphdb/tinkerpop/plugin/JanusGraphGremlinDriverPlugin.java]: Codacy Issue found: [Avoid throwing raw exception types.](https://app.codacy.com/app/JanusGraph/janusgraph/pullRequest?prid=3377550)

File: janusgraph-core/src/main/java/org/janusgraph/core/attribute/GeoshapeSerializer.java (hunk 7 above)
Comment (FlorianHockmann) [inline janusgraph-core/src/main/java/org/janusgraph/core/attribute/GeoshapeSerializer.java]: Codacy Issue found: [Avoid throwing raw exception types.](https://app.codacy.com/app/JanusGraph/janusgraph/pullRequest?prid=3377550)

File: janusgraph-core/src/main/java/org/janusgraph/graphdb/relations/RelationIdentifierUtils.java (hunk 3 above)
Comment (FlorianHockmann) [inline janusgraph-core/src/main/java/org/janusgraph/graphdb/relations/RelationIdentifierUtils.java]: Codacy Issue found: [Avoid reassigning parameters such as 'v'](https://app.codacy.com/app/JanusGraph/janusgraph/pullRequest?prid=3377550)

File: janusgraph-core/src/main/java/org/janusgraph/graphdb/transaction/StandardJanusGraphTx.java (hunk 5 above)
Comment (FlorianHockmann) [inline janusgraph-core/src/main/java/org/janusgraph/graphdb/transaction/StandardJanusGraphTx.java]: Codacy Issue found: [Avoid unused imports such as 'org.janusgraph.graphdb.relations'](https://app.codacy.com/app/JanusGraph/janusgraph/pullRequest?prid=3377550)

File: janusgraph-core/src/main/java/org/janusgraph/graphdb/relations/RelationIdentifierUtils.java (hunk 4 above)
Comment (FlorianHockmann) [inline janusgraph-core/src/main/java/org/janusgraph/graphdb/relations/RelationIdentifierUtils.java]: Codacy Issue found: [Avoid unused imports such as 'org.janusgraph.core'](https://app.codacy.com/app/JanusGraph/janusgraph/pullRequest?prid=3377550)

File: janusgraph-core/src/main/java/org/janusgraph/core/attribute/GeoshapeSerializer.java (hunk 8 above)
Comment (FlorianHockmann) [inline janusgraph-core/src/main/java/org/janusgraph/core/attribute/GeoshapeSerializer.java]: Codacy Issue found: [Switch statements should have a default label](https://app.codacy.com/app/JanusGraph/janusgraph/pullRequest?prid=3377550)

File: janusgraph-core/src/main/java/org/janusgraph/core/attribute/GeoshapeSerializer.java (hunk 1 above)
Comment (FlorianHockmann) [inline janusgraph-core/src/main/java/org/janusgraph/core/attribute/GeoshapeSerializer.java]: Codacy Issue found: [Avoid throwing raw exception types.](https://app.codacy.com/app/JanusGraph/janusgraph/pullRequest?prid=3377550)

File: janusgraph-core/src/main/java/org/janusgraph/core/attribute/GeoshapeSerializer.java (hunk 6 above)
Comment (FlorianHockmann) [inline janusgraph-core/src/main/java/org/janusgraph/core/attribute/GeoshapeSerializer.java]: Codacy Issue found: [Avoid reassigning parameters such as 'value'](https://app.codacy.com/app/JanusGraph/janusgraph/pullRequest?prid=3377550)

**Comment by FlorianHockmann** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
+ Solved 2
- Added 6
           

Complexity increasing per file
==============================
- janusgraph-driver/src/test/java/org/janusgraph/graphdb/tinkerpop/JanusGraphSerializerBaseIT.java  1
- janusgraph-driver/src/main/java/org/janusgraph/graphdb/tinkerpop/plugin/JanusGraphGremlinDriverPlugin.java  3
- janusgraph-driver/src/test/java/org/janusgraph/graphdb/tinkerpop/JanusGraphIoRegistryTest.java  2
- janusgraph-driver/src/test/java/org/janusgraph/graphdb/tinkerpop/io/graphson/JanusGraphSONModuleTest.java  2
- janusgraph-driver/src/main/java/org/janusgraph/graphdb/tinkerpop/JanusGraphIoRegistryV1d0.java  1
- janusgraph-driver/src/test/java/org/janusgraph/core/JanusGraphContainer.java  1
- janusgraph-driver/src/main/java/org/janusgraph/graphdb/relations/RelationIdentifier.java  9
- janusgraph-driver/src/main/java/org/janusgraph/core/attribute/Cmp.java  2
- janusgraph-core/src/main/java/org/janusgraph/graphdb/relations/RelationIdentifierUtils.java  10
- janusgraph-core/src/main/java/org/janusgraph/core/attribute/GeoshapeSerializer.java  18
- janusgraph-driver/src/main/java/org/janusgraph/graphdb/tinkerpop/JanusGraphIoRegistry.java  1
- janusgraph-driver/src/main/java/org/janusgraph/core/attribute/Geoshape.java  7
- janusgraph-core/src/main/java/org/janusgraph/graphdb/database/serialize/InternalAttributeUtil.java  1
- janusgraph-driver/src/main/java/org/janusgraph/graphdb/database/serialize/AttributeUtils.java  12
- janusgraph-driver/src/test/java/org/janusgraph/graphdb/tinkerpop/io/graphson/JanusGraphSONModuleIT.java  1
- janusgraph-driver/src/test/java/org/janusgraph/graphdb/tinkerpop/plugin/JanusGraphGremlinDriverPluginTest.java  3
- janusgraph-driver/src/test/java/org/janusgraph/graphdb/tinkerpop/JanusGraphGryoIT.java  1
- janusgraph-core/src/main/java/org/janusgraph/graphdb/query/JanusGraphPredicateUtils.java  14
- janusgraph-driver/src/test/java/org/janusgraph/core/attribute/GeoshapeHelperTest.java  1
- janusgraph-driver/src/test/java/org/janusgraph/util/encoding/LongEncodingTest.java  3
- janusgraph-driver/src/test/java/org/janusgraph/core/attribute/TinkerPopTextWrappingPredicateTest.java  1
         

Complexity decreasing per file
==============================
+ janusgraph-core/src/main/java/org/janusgraph/graphdb/tinkerpop/plugin/JanusGraphGremlinPlugin.java  -2
         

Clones added
============
- janusgraph-driver/src/test/java/org/janusgraph/graphdb/tinkerpop/JanusGraphIoRegistryTest.java  1
- janusgraph-driver/src/test/java/org/janusgraph/graphdb/tinkerpop/io/graphson/JanusGraphSONModuleTest.java  1
- janusgraph-driver/src/main/java/org/janusgraph/graphdb/relations/RelationIdentifier.java  2
- janusgraph-driver/src/test/java/org/janusgraph/graphdb/tinkerpop/plugin/JanusGraphGremlinDriverPluginTest.java  1
- janusgraph-driver/src/test/java/org/janusgraph/core/attribute/GeoshapeHelperTest.java  2
         
```


See the complete overview on [Codacy](https://app.codacy.com/app/JanusGraph/janusgraph/pullRequest?prid=3377550&bid=12157851)

File: janusgraph-driver/src/main/java/org/janusgraph/graphdb/tinkerpop/plugin/JanusGraphGremlinDriverPlugin.java (hunk 2 above)
Comment (FlorianHockmann) [inline janusgraph-driver/src/main/java/org/janusgraph/graphdb/tinkerpop/plugin/JanusGraphGremlinDriverPlugin.java]: Codacy Issue found: [Avoid throwing raw exception types.](https://app.codacy.com/app/JanusGraph/janusgraph/pullRequest?prid=3377550)

File: janusgraph-core/src/main/java/org/janusgraph/core/attribute/GeoshapeSerializer.java (hunk 6 above)
Comment (FlorianHockmann) [inline janusgraph-core/src/main/java/org/janusgraph/core/attribute/GeoshapeSerializer.java]: Codacy Issue found: [Avoid reassigning parameters such as 'value'](https://app.codacy.com/app/JanusGraph/janusgraph/pullRequest?prid=3377550)

File: janusgraph-core/src/main/java/org/janusgraph/core/attribute/GeoshapeSerializer.java (hunk 8 above)
Comment (FlorianHockmann) [inline janusgraph-core/src/main/java/org/janusgraph/core/attribute/GeoshapeSerializer.java]: Codacy Issue found: [Switch statements should have a default label](https://app.codacy.com/app/JanusGraph/janusgraph/pullRequest?prid=3377550)

File: janusgraph-core/src/main/java/org/janusgraph/graphdb/relations/RelationIdentifierUtils.java [hunk 9]
```diff
@@ -0,0 +1,93 @@
...
+
+    private static Iterable<? extends JanusGraphRelation> getJanusGraphRelations(RelationIdentifier rId, JanusGraphTransaction tx, JanusGraphVertex v, RelationType typeVertex) {
+        if (typeVertex.isEdgeLabel()) {
+            return findEdgeRelations(v, typeVertex, rId, tx);
+        } else {
+            return ((VertexCentricQueryBuilder) v.query()).noPartitionRestriction().types((PropertyKey) typeVertex).properties();
+        }
+    }
+
+    public static Iterable<? extends JanusGraphRelation> findEdgeRelations(JanusGraphVertex v, RelationType type, RelationIdentifier rId, JanusGraphTransaction tx){
```
Comment (FlorianHockmann) [inline janusgraph-core/src/main/java/org/janusgraph/graphdb/relations/RelationIdentifierUtils.java]: Codacy Issue found: [Avoid reassigning parameters such as 'v'](https://app.codacy.com/app/JanusGraph/janusgraph/pullRequest?prid=3377550)

File: janusgraph-core/src/main/java/org/janusgraph/core/attribute/GeoshapeSerializer.java (hunk 1 above)
Comment (FlorianHockmann) [inline janusgraph-core/src/main/java/org/janusgraph/core/attribute/GeoshapeSerializer.java]: Codacy Issue found: [Avoid throwing raw exception types.](https://app.codacy.com/app/JanusGraph/janusgraph/pullRequest?prid=3377550)

File: janusgraph-core/src/main/java/org/janusgraph/core/attribute/GeoshapeSerializer.java (hunk 7 above)
Comment (FlorianHockmann) [inline janusgraph-core/src/main/java/org/janusgraph/core/attribute/GeoshapeSerializer.java]: Codacy Issue found: [Avoid throwing raw exception types.](https://app.codacy.com/app/JanusGraph/janusgraph/pullRequest?prid=3377550)

**Comment by farodin91** [comment]
@porunov I merged it locally using fast forward

**Comment by farodin91** [comment]
b96aa6d26f74d3dc8f7404212bddeefe2d0790b4

**Pull request closed without merging**